apps/src/dance/ai/DanceAiClient.ts

```diff
@@ -40,66 +42,62 @@ const CachedPalettes: CachedWeightsMapping = UntypedCachedPalettes;
  * }
  */
 export function chooseEffects(
-  emojis: string[],
+  selectedEmojis: string[],
   quality: ChooseEffectsQuality
 ): GeneratedEffect {
   // Obtain final summed output weight based off input received
-  const outputTypes: CachedWeightsMapping[] = [
-    CachedBackgrounds,
-    CachedForegrounds,
-    CachedPalettes,
-  ];
-  const outputWeights: number[][] = outputTypes.map(function (map) {
-    return calculateOutputWeightsVector(emojis, map);
-  });
-
-  // Sort and slice top scoring options, mapped to their output identifiers (e.g. [[0.25, 'squiggles'], ...])
+  const cachedWeightsMappings: {[key: string]: CachedWeightsMapping} = {
+    backgroundEffect: CachedBackgroundEffects,
+    foregroundEffect: CachedForegroundEffects,
+    backgroundColor: CachedPalettes,
+  };
+  const outputWeightsForSelectedEmojis: {[key: string]: number[]} = {};
+  for (const output in cachedWeightsMappings) {
+    outputWeightsForSelectedEmojis[output] = calculateOutputWeightsVector(
+      selectedEmojis,
+      cachedWeightsMappings[output]
+    );
+  }
```

Comment:
not blocking at all, but I'd tend to do something like this in a single statement with a reduce:
```
const outputWeightsForSelectedEmojis: {[key: string]: number[]} =
  Object.entries(cachedWeightsMappings).reduce( (bucket, [key, cachedMapping]) => (
    { ...bucket, [key] : calculateOutputWeightsVector(selectedEmojis, cachedMapping) }
  ), {} )
```
(_not_ tested, so there may be typos)

Reply:
I'd like to keep things as easy to understand for future readers not familiar with context so I think I may keep as is for that reason.